src/tox/session.py

```diff
@@ -592,9 +600,49 @@ def subcommand_test(self):
                         self.installpkg(venv, venv.package)
 
                 self.runenvreport(venv)
-                self.runtestenv(venv)
-        retcode = self._summary()
-        return retcode
+            self.runtestenv(venv)
+
+    def run_parallel(self):
+        """here we'll just start parallel sub-processes"""
+        live_out = self.config.option.parallel_live
+        args = [sys.executable, "-m", "tox"] + self.config.args
+        try:
+            position = args.index("--")
+        except ValueError:
+            position = len(args)
+        try:
+            parallel_at = args[0:position].index("--parallel")
+            del args[parallel_at]
+            position -= 1
+        except ValueError:
+            pass
+
+        tox_runs = {}
+        for venv in self.venvlist:
+            env = os.environ.copy()
+            env["_PARALLEL_TOXENV"] = venv.envconfig.envname
+            args_sub = list(args)
+            if hasattr(venv, "package"):
+                args_sub.insert(position, str(venv.package))
+                args_sub.insert(position, "--installpkg")
+            stdout, stderr = (None, None) if live_out else (subprocess.PIPE, subprocess.PIPE)
+            run = subprocess.Popen(args_sub, env=env, stdout=stdout, stderr=stderr)
+            tox_runs[venv.name] = (venv, run)
+
+        to_finish = set(tox_runs.keys())
```

Comment:
`to_finish` reads a bit strage to me here. How about `not_done` or `still_running`?

Reply:
maybe ``not_done`` as some might still be waiting to be queued with the queuing logic 👍